lib/spack/spack/solver/asp.py

```diff
@@ -1654,6 +1656,12 @@ def _facts_from_concrete_spec(self, spec, possible):
 
             # add OS to possible OS's
             for dep in spec.traverse():
+                self.possible_versions[dep.name].add(dep.version)
```

Comment:
The comment above this block probably should change to reflect the new content, maybe

```
# ensure all attributes of installed specs are considered possible values.

Reply:
Fixed in 7727cd1